src/controllers/productController.js

```diff
@@ -25,95 +25,69 @@ class productController{
         }
     }
 
-  static async createProduct(req, res) {
-    try {
-
-      if (req.user.role.roleName !== "vendor") {
-        return res
-          .status(401)
-          .json({ status: "fail", message: req.t("Unauthorized") });
-      };
-
-     const {
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        CategoryId,
-        expiredDate
-      } = req.body;
-      const productImage = req.files.map((img) => img.path);
-      const product = await Product.create({
-        vendorId: req.user.role.id,
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        expiredDate,
-        CategoryId,
-        productImage
-      });
-
-      await product.save();
-      return res
-        .status(200)
-        .json({
+    static async createProduct(req, res) {
+      if (req.user.roleName === "vendor") {
+        try {
+          const {
+            productName,
+            productPrice,
+            quantity,
+            productDescription,
+            productOwner,
+            CategoryId,
+            bonus,
+            expiredDate
+          } = req.body;
+          const productImage = req.files.map((img) => img.path);
+          const postProduct = await Product.create({
+            vendorId: req.user.id,
+            productName,
+            productPrice,
+            quantity,
+            productDescription,
+            productOwner,
+            expiredDate,
+            CategoryId,
+            bonus,
+            productImage
+          });
+          return res.status(200).json({
           status: "success",
           message: "product created",
-          product: { product }
+          product:postProduct 
         });
-    } catch (error) {
-      return res.json({ error: error.message });
+      } catch (error) {
+        return res.json({ error: error.message });
+      }
+    } else {
+      return res
+        .status(401)
+        .json({status: "fail", message: req.t("Unauthorized") });
     }
   }
+  
 
   static async categoryController(req, res) {
     const { name } = req.body;
     try {
 
-      if (req.user.role.roleName !== "vendor") {
+      if (req.user.roleName !== "vendor") {
         return res
           .status(401)
           .json({ status: "fail", message: req.t("Unauthorized") });
       }
 
 
-      const category = await Category.create({
+      const postCategory = await Category.create({
         name
       });
 
```

Comment:
- stay consistent with responses e.g 

`res.status(636).json({status: "success", message: ""})`
`res.status(636).json({status: "fail", error: ""})`

- also correct status code for created is `201`